src/sentry/releasehealth/metrics.py

```diff
@@ -340,3 +346,72 @@ def _count_users(total: bool, referrer: str) -> Dict[Any, int]:
             rv[project_id, release] = adoption
 
         return rv
+
+    def check_has_health_data(
+        self, projects_list: Sequence[Union[ProjectId, ProjectRelease]]
+    ) -> Set[Union[ProjectId, ProjectRelease]]:
+        now = datetime.now(pytz.utc)
+        start = now - timedelta(days=3)
+
+        projects_list = list(projects_list)
+
+        if len(projects_list) == 0:
+            return set()
+
+        includes_releases = type(projects_list[0]) == tuple
+
+        if includes_releases:
+            project_ids = [x[0] for x in projects_list]  # type: ignore
+        else:
+            project_ids = [x for x in projects_list]
+
+        org_id = self._get_org_id(project_ids)
+
+        where_clause = [
+            Condition(Column("org_id"), Op.EQ, org_id),
+            Condition(Column("project_id"), Op.IN, project_ids),
+            Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session")),
+            Condition(Column("timestamp"), Op.GTE, start),
+            Condition(Column("timestamp"), Op.LT, now),
+        ]
+
+        if includes_releases:
+            releases = [x[1] for x in projects_list]  # type: ignore
+            release_column_name = tag_key(org_id, "release")
+            releases_ids = [
+                release_id
+                for release_id in [try_get_tag_value(org_id, release) for release in releases]
+                if release_id is not None
+            ]
+            where_clause.append(Condition(Column(release_column_name), Op.IN, releases_ids))
+            column_names = ["project_id", release_column_name]
+
+            def extract_raw_info(
+                row: Mapping[str, Union[int, str]]
+            ) -> Union[ProjectId, ProjectRelease]:
+                return row.get("project_id"), reverse_tag_value(org_id, row.get(release_column_name))  # type: ignore
```

Comment:
I wonder if we should also return an entry for projects which don't have any data (either because the indexer failed to resolve the release version, or because there haven't been any sessions for this release). How does the sessions backend respond in such a case?

Reply:
Session backend only returns what it finds and since release is a string column in the sessions table there is no lookup for indexes for either the "release" tag or the specific release entry.

IMO I think we should just skip the result if we can't find the appropriate data (although not finding the index for a stored release name would be an error in how the indexer works or how the data was inserted).